cylc/flow/task_pool.py

```diff
@@ -974,76 +1046,66 @@ def check_abort_on_task_fails(self):
             )
             for itask in self.get_tasks())
 
-    def match_dependencies(self):
-        """Run time dependency negotiation.
-
-        Tasks attempt to get their prerequisites satisfied by other tasks'
-        outputs. Brokered negotiation is O(n) in number of tasks.
+    def spawn(self, up_name, up_point, name, point, message=None,
+              finished=False, go=False):
+        """Spawn a new task proxy and/or update its prerequisites.
 
         """
-        all_task_outputs = set()
-        for itask in self.get_tasks():
-            for output in itask.state.outputs.get_completed():
-                all_task_outputs.add((itask.tdef.name,
-                                      str(itask.point),
-                                      output))
-        for itask in self.get_tasks():
-            # Try to satisfy itask if not already satisfied.
-            if itask.state.prerequisites_are_not_all_satisfied():
-                itask.state.satisfy_me(all_task_outputs)
-
-    def force_spawn(self, itask):
-        """Spawn successor of itask."""
-        if itask.has_spawned:
-            return None
-        itask.has_spawned = True
-        LOG.debug('[%s] -forced spawning', itask)
-        next_point = itask.next_point()
-        if next_point is None:
+        if self.config.initial_point and point < self.config.initial_point:
```

Comment:
Hmm, looking back, we have this on master:
```python
       if not self.config.initial_point and not self.is_restart:
            LOG.warning('No initial cycle point provided - no cycling tasks '
                        'will be loaded.')
       # (does not abort here...)
```
But I don't think this can actually happen (still investigating)

Reply:
Although `git blame` attributes that to Matt, relatively recently. :thinking:

Reply:
I'll add a TODO comment to investigate and remove this check if possible, as it goes beyond this branch and the code above (in master too) suggests no ICP is possible somehow.

Reply:
Didn't check to see it was the same before, doesn't need investigation on this branch.

Reply:
Comment added anyway.